v2/pkg/subscraping/sources/netlas/netlas.go

```diff
@@ -0,0 +1,196 @@
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/projectdiscovery/subfinder/v2/pkg/subscraping"
+)
+
+type Item struct {
+	Data struct {
+		A           []string `json:"a,omitempty"`
+		Txt         []string `json:"txt,omitempty"`
+		LastUpdated string   `json:"last_updated,omitempty"`
+		Timestamp   string   `json:"@timestamp,omitempty"`
+		Ns          []string `json:"ns,omitempty"`
+		Level       int      `json:"level,omitempty"`
+		Zone        string   `json:"zone,omitempty"`
+		Domain      string   `json:"domain,omitempty"`
+		Cname       []string `json:"cname,omitempty"`
+		Mx          []string `json:"mx,omitempty"`
+	} `json:"data"`
+}
+
+type Response struct {
+	Items []Item `json:"items"`
+	Took  int    `json:"took"`
+}
+
+type DomainsCountResponse struct {
+	Count int `json:"count"`
+}
+
+// Source is the passive scraping agent
+type Source struct {
+	apiKeys   []string
+	timeTaken time.Duration
+	errors    int
+	results   int
+	skipped   bool
+}
+
+// func get_domains(offset string, domain string) int {
+
+// 	return resp.StatusCode
+// }
+
+func (s *Source) Run(ctx context.Context, domain string, session *subscraping.Session) <-chan subscraping.Result {
+	results := make(chan subscraping.Result)
+	s.errors = 0
+	s.results = 0
+
+	go func() {
+		defer func(startTime time.Time) {
+			s.timeTaken = time.Since(startTime)
+			close(results)
+		}(time.Now())
+
+		// To get count of domains
+		endpoint := "https://app.netlas.io/api/domains_count/"
+		paramss := url.Values{}
+		paramss.Set("q", "domain:(domain:*."+domain+" "+"AND NOT domain:"+domain+")")
+		countUrl := endpoint + "?" + paramss.Encode()
+
+		client := &http.Client{}
+		req, _ := http.NewRequest("GET", countUrl, nil)
+		req.Header.Set("accept", "application/json")
+
+		// Pick an API key
+		randomApiKey := subscraping.PickRandom(s.apiKeys, s.Name())
+		if randomApiKey != "" {
+			req.Header.Set("X-API-Key", randomApiKey)
+		}
+
+		resp, err := client.Do(req)
+		if err != nil {
+			results <- subscraping.Result{Source: s.Name(), Type: subscraping.Error, Error: err}
+			s.errors++
+			return
+		} else if resp.StatusCode != 200 {
+			results <- subscraping.Result{Source: s.Name(), Type: subscraping.Error, Error: fmt.Errorf("request rate limited with status code %d", resp.StatusCode)}
+			s.errors++
+			return
+		}
+		defer resp.Body.Close()
+
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			results <- subscraping.Result{Source: s.Name(), Type: subscraping.Error, Error: fmt.Errorf("Error Reading ressponse body")}
+			s.errors++
+			return
+		}
+
+		// Parse the JSON response
+		var domainsCount DomainsCountResponse
+		err = json.Unmarshal(body, &domainsCount)
+		if err != nil {
+			results <- subscraping.Result{Source: s.Name(), Type: subscraping.Error, Error: err}
+			s.errors++
+			return
+		}
+
+		//Define the API endpoint URL and query parameters
+
+		for i := 0; i < domainsCount.Count; i += 20 {
+
+			time.Sleep(1200 * time.Millisecond)
```

Comment:
I noticed that the pricing page indicates a request rate of 1/s, but I see a rate of 1.2/s. Can you help me understand the reason for this difference? Thanks!

Reply:
I don't remember why I did this but now as per my testing 1/s is working fine I have the necessary changes.